pkg/consolegraphql/resolvers/resolver_addressspace.go

```diff
@@ -200,7 +200,24 @@ func (r *addressSpaceK8sResolver) Addresses(ctx context.Context, obj *consolegra
 }
 
 func (r *queryResolver) MessagingCertificateChain(ctx context.Context, input v1.ObjectMeta) (string, error) {
-	panic("implement me")
+	requestState := server.GetRequestStateFromContext(ctx)
+	viewFilter := requestState.AccessController.ViewFilter()
+
+	filter := fmt.Sprintf("`$.ObjectMeta.Name` = '%s' and `$.ObjectMeta.Namespace` = '%s'", input.Name, input.Namespace)
+
+	fltrfunc, err := BuildFilter(&filter)
+	if err != nil {
+		return "", err
+	}
+	objects, e := r.Cache.Get(cache.PrimaryObjectIndex, "AddressSpace/", cache.And(viewFilter, fltrfunc))
```

Comment:
if you pass the key formed by "AddressSpace/<namespace>/<name>, you avoid the need to walk the whole tree of addressspaces.